src/workerd/server/server.c++

```diff
@@ -904,6 +905,158 @@ kj::Own<Server::Service> Server::makeDiskDirectoryService(
+    // about the implementation.
+    if (method != kj::HttpMethod::GET) {
+      return response.sendError(501, "Unsupported Operation", responseHeaders);
+    }
+
+    if (url.endsWith("/json/version")) {
+      responseHeaders.set(kj::HttpHeaderId::CONTENT_TYPE, "application/json"_kj);
+      auto content = kj::str("{\"Browser\": \"workerd\", \"Protocol-Version\": \"1.3\" }");
+      auto out = response.send(200, "OK", responseHeaders, content.size());
+      return out->write(content.begin(), content.size()).attach(kj::mv(content), kj::mv(out));
+    } else if (url.endsWith("/json") || url.endsWith("/json/list")) {
+      responseHeaders.set(kj::HttpHeaderId::CONTENT_TYPE, "application/json"_kj);
+
+      auto baseWsUrl = KJ_UNWRAP_OR(headers.get(kj::HttpHeaderId::HOST), {
+        return response.sendError(400, "Bad Request", responseHeaders);
+      });
+
+      kj::Vector<kj::String> entries(isolates.size());
+      for (auto& entry : isolates) {
+        kj::Vector<kj::String> fields(9);
+        fields.add(kj::str("\"id\":\"", entry.key ,"\""));
+        fields.add(kj::str("\"title\":\"workerd: worker ", entry.key ,"\""));
+        fields.add(kj::str("\"type\":\"node\""));
+        fields.add(kj::str("\"description\":\"workerd worker\""));
+        fields.add(kj::str("\"webSocketDebuggerUrl\":\"ws://",
+                            baseWsUrl ,"/", entry.key ,"\""));
+        fields.add(kj::str("\"devtoolsFrontendUrl\":\"devtools://devtools/bundled/js_app.html?experiments=true&v8only=true&ws=", baseWsUrl ,"/\""));
+        fields.add(kj::str("\"devtoolsFrontendUrlCompat\":\"devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=", baseWsUrl ,"/\""));
+        fields.add(kj::str("\"faviconUrl\":\"https://workers.cloudflare.com/favicon.ico\""));
+        fields.add(kj::str("\"url\":\"https://workers.dev\""));
+        entries.add(kj::str('{', kj::strArray(fields, ",") ,'}'));
+      }
+      auto content = kj::str('[', kj::strArray(entries, ","), ']');
+
+      auto out = response.send(200, "OK", responseHeaders, content.size());
+      return out->write(content.begin(), content.size()).attach(kj::mv(content), kj::mv(out));
+    }
+
+    return response.sendError(500, "Not yet implemented", responseHeaders);
+  }
+
+  kj::Promise<void> listen(kj::Own<kj::ConnectionReceiver> listener) {
+    return server.listenHttp(*listener).attach(kj::mv(listener));
+  }
+
+  class Handle final {
+    // Holds the registration of a Worker::Isolate with the InspectorService.
+  public:
+    Handle(kj::String name, InspectorService& service, Worker::Isolate* isolate)
+        : name(kj::mv(name)), service(service) {
+      service.isolates.insert(kj::str(name), isolate);
+    }
+
+    ~Handle() { service.isolates.erase(name); }
+
+  private:
+    kj::String name;
+    InspectorService& service;
+
+    friend class InspectorService;
+  };
+
+  kj::Own<Handle> getHandle(kj::StringPtr name, Worker::Isolate* isolate) {
```

Comment:
Nitpick: `getHandle()` makes it sound like creating a handle is the purpose of the function. I'd go with `registerIsolate()`.

(I guess I'd rename `Handle` to `Registration`, too.)